02/05/17 William Benjamin Shields

This is City of Seattle Wages Compared by Gender data from a CSV file. Dataset: https://data.seattle.gov/City-Business/City-of-Seattle-Wages-Comparison-by-Gender-All-Job/cf52-s8er

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

**Dataset:**

In [3]:
df = pd.read_csv("City_of_Seattle_Wages___Comparison_by_Gender_-_All_Job_Classifications.csv", sep=",", header=None)

In [4]:
df = df.rename(columns={0: "Jobtitle", 1: "Female Avg Hrly Rate", 2: "No. Female Empl", 3: "Average of Female MONTHS LONGEVITY IN CURRENT CLASSIFICATION", 4: "Male Avg Hrly Rate", 5: "No. Male Empl", 6: "Average of Male MONTHS LONGEVITY IN CURRENT CLASSIFICATION", 7: "Total Avg Hrly Rate", 8: "Total No. Empl", 9: "Total Average of MONTHS LONGEVITY IN CURRENT CLASSIFICATION", 10: "Ratio of women's hourly rate to men's hourly rate", 11: "Notes"}).drop(0, axis=0)
df

Jobtitle Female Avg Hrly Rate No. Female Empl  \
1                     Accountant                30.58              23   
2                Accountant,Prin                38.75              23   
3                  Accountant,Sr                34.48              23   
4                       Act Exec                 43.1               4   
5                    Actg Tech I                22.56               1   
6                 Actg Tech I-BU                22.56              10   
7                   Actg Tech II                23.98               7   
8                Actg Tech II-BU                24.19              74   
9                  Actg Tech III                26.64               7   
10              Actg Tech III-BU                26.69              32   
11               Actg Tech II-MC                24.35               2   
12             Actg Tech Supv-BU                28.84               4   
13                  Admin Spec I                22.35              15   
14               Admin Spec I-BU                22.43              78   
15                 Admin Spec II                24.32              28   
16              Admin Spec II-BU                24.25             140   
17                Admin Spec III                 26.1              12   
18             Admin Spec III-BU                25.94              30   
19             Admin Spec III-MC                26.26               1   
20              Admin Spec II-MC                24.35              28   
21               Admin Spec I-MC                22.56              16   
22            Admin Staff Anlyst                35.59              40   
23              Admin Staff Asst                30.86              54   
24            Admin Support Asst                20.61               1   
25         Admin Support Asst-BU                 20.3              10   
26         Admin Support Asst-MC                  NaN             NaN   
27         Admin Support Supv-BU                27.58              10   
28         Admin Support Supv-MC                27.78               4   
29             Administrator-Law                48.11               1   
30                 Adms Employee                17.11               5   
..                           ...                  ...             ...   
843        Wtr Meter Repairer,Sr                27.67               1   
844                  Wtr Pipe CC                  NaN             NaN   
845           Wtr Pipe CC-WDM II                  NaN             NaN   
846                 Wtr Pipe Wkr                30.18               2   
847       Wtr Pipe Wkr Sr-WDM II                33.64               4   
848              Wtr Pipe Wkr,Sr                  NaN             NaN   
849           Wtr Pipe Wkr-WDM I                30.46               1   
850           Wtr Quality Anlyst                33.69               3   
851      Wtr Quality Anlyst,Asst                31.16               2   
852        Wtr Quality Anlyst,Sr                 36.3               2   
853      Wtr Quality Anlyst,Supv                 39.2               1   
854             Wtr Quality Engr                44.74               1   
855          Wtr Quality Engr,Sr                  NaN             NaN   
856                  Wtr Syst Op                  NaN             NaN   
857               Wtr Syst Op,Sr                41.72               1   
858               Wtr Systs Supv                  NaN             NaN   
859             Wtr Treatment CC                 36.3               1   
860     Wtr Treatment Equip Tech                  NaN             NaN   
861          Wtr Treatment Op,Sr                  NaN             NaN   
862  Wtr Treatment Op-Asg Relief                  NaN             NaN   
863       Wtr Treatment Op-WDM I                28.22               2   
864            Wtrshed Inspector                  NaN             NaN   
865         Wtrshed Inspector,Sr                34.94               1   
866      

**QUESTION: What is the ratio of female to male employees working for the City of Seattle?**

In [5]:
num_females = df.loc[872]["No. Female Empl"]
num_females

'3600'

In [6]:
num_males = df.loc[872]["No. Male Empl"]
num_males

'6285'

In [7]:
ratio = int(num_females) / int(num_males)
ratio

0.5727923627684964

**ANSWER: There are almost twice as many males as females working for the City of Seattle.**

**QUESTION: What percentage of job classifications only employ men? What percentage of job classifications only employ women?**

In [8]:
num_class = 872 - 1
num_class

871

In [9]:
jobs_no_men = df["Notes"].values == "no men in title"
num_jobs_no_men = np.cumsum(jobs_no_men)[-1]
num_jobs_no_men

144

In [10]:
jobs_no_women = df["Notes"].values == "no women in title"
num_jobs_no_women = np.cumsum(jobs_no_women)[-1]
num_jobs_no_women

294

In [11]:
percent_only_men = (num_jobs_no_women / num_class) * 100
percent_only_men

33.754305396096441

In [12]:
percent_only_women = (num_jobs_no_men / num_class) * 100
percent_only_women

16.532721010332953

**ANSWER: 33.7 percent of jobs only employ men. 16.5 percent of jobs only employ women. There are around twice as many jobs that employ only men as there are jobs that only employ women.**

**QUESTION: Of these jobs, how many employ fewer than five people?**

In [13]:
class_no_men = df.loc[df["Notes"] == "no men in title"]
class_no_men

Jobtitle Female Avg Hrly Rate No. Female Empl  \
5                  Actg Tech I                22.56               1   
9                Actg Tech III                26.64               7   
11             Actg Tech II-MC                24.35               2   
12           Actg Tech Supv-BU                28.84               4   
19           Admin Spec III-MC                26.26               1   
24          Admin Support Asst                20.61               1   
29           Administrator-Law                48.11               1   
34     Animal Contrl Ofcr Supv                30.72               2   
36        Apprenticeship Coord                33.69               2   
40    Aquarium Laboratory Spec                26.26               1   
48              Arts Prgm Spec                32.35               5   
63               Bailiff,Chief                29.41               1   
64               Benefits Asst                26.26               2   
70      Bldg Plans Examiner,Sr                42.25               2   
82    Bridge Opns General Supv                37.39               1   
84               Camp Svc Aide                17.99               2   
85            Camp Svc Aide,Sr                 19.8               1   
89    Capital Prjts Coord,Asst                 38.5               3   
90   Capital Prjts Coord,Chief                49.33               1   
99       Cblspl CC-Asg Locator                46.28               1   
107        City Archivist,Asst                38.48               3   
128          Claims Adjuster I                31.76               1   
130     Code Compliance Anlyst                 37.7               2   
131      Code Compliance Coord                 43.1               1   
132         Code Dev Anlyst,Sr                40.74               1   
144            Contract Anlyst                31.18               3   
149              Council Clerk                31.76               2   
155           Court Clerk Supv                 26.8               3   
156    Court Interpreter Coord                 39.2               1   
163            Cust Svc Rep,Sr                26.18               5   
..                         ...                  ...             ...   
728          StratAdvsr1,CL&PS                44.84               2   
736   StratAdvsr1,Info Technol                45.22               1   
738          StratAdvsr1,PC&RM                47.76               2   
751     StratAdvsr2,Human Svcs                45.61               6   
753           StratAdvsr2,P&FM                47.26               2   
760      StratAdvsr3,Exempt-BU                50.01               1   
763   StratAdvsr3,Info Technol                53.96               1   
773         Strucl Iron Wkr CC                46.28               1   
782           Tax Auditor,Asst                31.16               1   
783             Tax Auditor,Sr                36.99               1   
786          Temp Emplmnt Spec                33.69               1   
788              Tenprint Supv                35.57               2   
789         Title Records Tech                22.56               1   
798        Treasury Cashier,Sr                34.94               1   
802        Trng Dev&Prgm Coord                33.69               1   
804         Trng&Ed Coord,Asst                26.79               4   
805      Trng&Ed Coord,Asst-BU                26.79               1   
809                 Usher,Head                20.81               1   
815             Util Acts Supv                31.91               5   
816           Util Astnce Supv                33.69               1   
819          Util Hearing Ofcr                42.25               2   
827       Victim Advocate Supv                 36.3               1   
830             Volunteer Supv                26.26               1   
836       Workers' Comp Anlyst                35.26               5   
837    Workers' Comp Anlyst,Sr                

In [19]:
no_men_low_num = class_no_men.loc[df["Total No. Empl"] < "5"].count()
no_men_low_num

Jobtitle                                                        135
Female Avg Hrly Rate                                            135
No. Female Empl                                                 135
Average of Female MONTHS LONGEVITY IN CURRENT CLASSIFICATION    135
Male Avg Hrly Rate                                                0
No. Male Empl                                                     0
Average of Male MONTHS LONGEVITY IN CURRENT CLASSIFICATION        0
Total Avg Hrly Rate                                             135
Total No. Empl                                                  135
Total Average of MONTHS LONGEVITY IN CURRENT CLASSIFICATION     135
Ratio of women's hourly rate to men's hourly rate                 0
Notes                                                           135
dtype: int64

In [20]:
class_no_women = df.loc[df["Notes"] == "no women in title"]
class_no_women

Jobtitle Female Avg Hrly Rate No. Female Empl  \
26          Admin Support Asst-MC                  NaN             NaN   
31      Adms Personnel Dispatcher                  NaN             NaN   
35                   Appraiser,Sr                  NaN             NaN   
39           Aquarium Biologist 3                  NaN             NaN   
41        Aquarium Systs Op,Chief                  NaN             NaN   
45                       Arborist                  NaN             NaN   
46                   Architect,Sr                  NaN             NaN   
50                  Asphalt Raker                  NaN             NaN   
51               Asphalt Raker,Sr                  NaN             NaN   
52               Asst To The Supt                  NaN             NaN   
53             Auto Body Wkr/Pntr                  NaN             NaN   
54                      Auto Engr                  NaN             NaN   
55                   Auto Engr,Sr                  NaN             NaN   
56                Auto Equip Pntr                  NaN             NaN   
57                  Auto Maint CC                  NaN             NaN   
61           Auto Sheet Metal Wkr                  NaN             NaN   
65         Bldg Inspector,Journey                  NaN             NaN   
66      Bldg Inspector,Sr(Expert)                  NaN             NaN   
67          Bldg Inspector,Strucl                  NaN             NaN   
68            Bldg Operating Engr                  NaN             NaN   
71   Bldg/Facilities Opns Supv-BU                  NaN             NaN   
72    Bridge Carpentry&Maint Lead                  NaN             NaN   
73                Bridge Elecl CC                  NaN             NaN   
74        Bridge Elecl Maint Supv                  NaN             NaN   
75      Bridge Maint General Supv                  NaN             NaN   
76         Bridge Maint Mech Hlpr                  NaN             NaN   
77          Bridge Maint Mechanic                  NaN             NaN   
78       Bridge Maint Mechanic CC                  NaN             NaN   
81                 Bridge Opns CC                  NaN             NaN   
93                   Carpenter CC                  NaN             NaN   
..                            ...                  ...             ...   
787             Tennis Instructor                  NaN             NaN   
791     Traffic Sign&Marking CC I                  NaN             NaN   
799                  Tree Trimmer                  NaN             NaN   
800             Tree Trimmer,Lead                  NaN             NaN   
801             Tree Trimming Rep                  NaN             NaN   
812           Util Act Rep Supv I                  NaN             NaN   
817          Util Constr Lead Wkr                  NaN             NaN   
824         Util Systs Maint Tech                  NaN             NaN   
825      Util Systs Maint Tech,Sr                  NaN             NaN   
831                Warehouse Supv                  NaN             NaN   
833                 Warehouser,Sr                  NaN             NaN   
840           Wtr Laboratory Tech                  NaN             NaN   
841             Wtr Meter Changer                  NaN             NaN   
842            Wtr Meter Repairer                  NaN             NaN   
844                   Wtr Pipe CC                  NaN             NaN   
845            Wtr Pipe CC-WDM II                  NaN             NaN   
848               Wtr Pipe Wkr,Sr                  NaN             NaN   
855           Wtr Quality Engr,Sr                  NaN             NaN   
856                   Wtr Syst Op                  NaN             NaN   
858                Wtr Systs Supv                  NaN             NaN   
860      Wtr Treatment Equip Tech                  NaN             NaN   
861           Wtr Treatment Op,Sr                  NaN             NaN   
862   Wtr Treatment Op-Asg Re

In [21]:
no_women_low_num = class_no_women.loc[df["Total No. Empl"] < "5"].count()
no_women_low_num

Jobtitle                                                        258
Female Avg Hrly Rate                                              0
No. Female Empl                                                   0
Average of Female MONTHS LONGEVITY IN CURRENT CLASSIFICATION      0
Male Avg Hrly Rate                                              258
No. Male Empl                                                   258
Average of Male MONTHS LONGEVITY IN CURRENT CLASSIFICATION      258
Total Avg Hrly Rate                                             258
Total No. Empl                                                  258
Total Average of MONTHS LONGEVITY IN CURRENT CLASSIFICATION     258
Ratio of women's hourly rate to men's hourly rate                 0
Notes                                                           258
dtype: int64

In [23]:
ratio_high_num_no_men = (144 - 135) / 144
ratio_high_num_no_men

0.0625

In [24]:
ratio_high_num_no_women = (294 - 258) / 294
ratio_high_num_no_women

0.12244897959183673

**ANSWER: Of 144 jobs without men, 135 employ under five people. Of 294 jobs without women, 258 employ under five people. In other words, the vast majority of jobs that do not employ either women or men are composed of small teams, as would be expected. However, of these jobs, the ratio of jobs that employ only women that employ greater than five people is 6%. The ratio of jobs that employ only men that employ greater than five people is 12%. This strikes me as a statistically significant difference.**

**QUESTION: How many job classifications pay females higher than males? How many job classifications pay males higher than females?**

In [33]:
new_df = df.drop(872, axis=0)
num_females_more = new_df.loc[new_df["Ratio of women's hourly rate to men's hourly rate"] > "100"].count()
num_females_more

Jobtitle                                                        433
Female Avg Hrly Rate                                            433
No. Female Empl                                                 433
Average of Female MONTHS LONGEVITY IN CURRENT CLASSIFICATION    433
Male Avg Hrly Rate                                              433
No. Male Empl                                                   433
Average of Male MONTHS LONGEVITY IN CURRENT CLASSIFICATION      433
Total Avg Hrly Rate                                             433
Total No. Empl                                                  433
Total Average of MONTHS LONGEVITY IN CURRENT CLASSIFICATION     433
Ratio of women's hourly rate to men's hourly rate               433
Notes                                                             0
dtype: int64

In [36]:
new_df = df.drop(872, axis=0)
num_males_more = new_df.loc[new_df["Ratio of women's hourly rate to men's hourly rate"] < "100"].count()
num_males_more

Jobtitle                                                        0
Female Avg Hrly Rate                                            0
No. Female Empl                                                 0
Average of Female MONTHS LONGEVITY IN CURRENT CLASSIFICATION    0
Male Avg Hrly Rate                                              0
No. Male Empl                                                   0
Average of Male MONTHS LONGEVITY IN CURRENT CLASSIFICATION      0
Total Avg Hrly Rate                                             0
Total No. Empl                                                  0
Total Average of MONTHS LONGEVITY IN CURRENT CLASSIFICATION     0
Ratio of women's hourly rate to men's hourly rate               0
Notes                                                           0
dtype: int64

**ANSWER:**

**QUESTION:**